kie-remote/kie-remote-services/src/test/java/org/kie/remote/services/rest/query/data/QueryResourceQueryParametersTest.java

```diff
@@ -0,0 +1,277 @@
+        }
+
+        for( String param : regexParams ) {
+            if( param == null ) {
+                continue;
+            }
+            queryParams.clear();
+            String reParam = param + "_re";
+            allParams.add(reParam);
+            addParam(queryParams, reParam, "1*");
+            checkIfParametersAreAllowed(queryParams, allowedParams, true, "test");
+        }
+
+        for( String param : nameValueParams ) {
+            if( param == null ) {
+                continue;
+            }
+            queryParams.clear();
+            String nvParam = param + "_nameofvar";
+            addParam(queryParams, param + "_name", "value");
+            allParams.add(nvParam);
+            checkIfParametersAreAllowed(queryParams, allowedParams, true, "test");
+        }
+
+    }
+
+    @Test
+    public void testUniqueParameters() throws Exception {
+
+        List<Field> paramFields =
+                new LinkedList<Field>(Arrays.asList(QueryResourceData.class.getDeclaredFields()));
+        Iterator<Field> iter = paramFields.iterator();
+        List<String[]> allParams = new ArrayList<String[]>(paramFields.size());
+        while( iter.hasNext() ) {
+           Field field = iter.next();
+           if( field.getName().equals("minMaxParams")
+               || field.getName().equals("nameValueParams")
+               || field.getName().equals("regexParams") ) {
+               continue;
+           }
+           if( ! Modifier.isStatic(field.getModifiers()) ) {
+              iter.remove();
+              continue;
+           }
+           if( ! field.getType().equals(String[].class) ) {
+              continue;
+           }
+           if( ! field.getName().contains("Params") ) {
+              continue;
+           }
+           field.setAccessible(true);
+           allParams.add((String []) field.get(null));
+        }
+
+        assertFalse( "No params found", allParams.isEmpty() );
+
+        Set<String> params = new HashSet<String>();
+        for( String [] paramArr : allParams ) {
+            for( String param : paramArr ) {
+                if( param == null ) {
+                    continue;
+                }
+                assertTrue( "Param \"" + param + "\" contains uppercase letters",
+                        param.toLowerCase().equals(param) );
+                assertTrue( "Param \"" + param + "\" is used twice",
+                        params.add(param) );
+            }
+        }
+
+        for( String param : ResourceBase.paginationParams ){
+            assertTrue( "Param \"" + param + "\" contains uppercase letters",
+                    param.toLowerCase().equals(param) );
+            assertTrue( "Param \"" + param + "\" is used twice",
+                    params.add(param) );
+        }
+        assertTrue( "Param \"" +  PROC_INST_ID_PARAM_NAME + "\" is used twice",
+                params.add(PROC_INST_ID_PARAM_NAME) );
+    }
+
+    @Test
+    @Ignore
```

Comment:
Good point, I'll remove the `@Test` and `@Ignore`. 

(These are helper methods).